source/fe/mapping_q_generic.cc

```diff
@@ -306,39 +306,48 @@ namespace internal
       {
         Point<dim> p_unit;
 
-        dealii::FullMatrix<double>  KA(GeometryInfo<dim>::vertices_per_cell, dim);
-        dealii::Vector <double>  Kb(GeometryInfo<dim>::vertices_per_cell);
+        // use fixed size arrays due to the small dimensions
 
-        KA.fill( (double *)(TransformR2UInitialGuess<dim>::KA) );
-        for (unsigned int i=0; i<GeometryInfo<dim>::vertices_per_cell; ++i)
-          Kb(i) = TransformR2UInitialGuess<dim>::Kb[i];
-
-        FullMatrix<double> Y(spacedim, GeometryInfo<dim>::vertices_per_cell);
-        for (unsigned int v=0; v<GeometryInfo<dim>::vertices_per_cell; v++)
-          for (unsigned int i=0; i<spacedim; ++i)
-            Y(i,v) = vertex[v][i];
-
-        FullMatrix<double> A(spacedim,dim);
-        Y.mmult(A,KA); // A = Y*KA
-        dealii::Vector<double> b(spacedim);
-        Y.vmult(b,Kb); // b = Y*Kb
-
-        for (unsigned int i=0; i<spacedim; ++i)
-          b(i) -= p[i];
-        b*=-1;
-
-        dealii::Vector<double> dest(dim);
+        // A = vertex * KA
+        double A[spacedim][dim];
+        for (unsigned int d=0; d<spacedim; ++d)
+          for (unsigned int e=0; e<dim; ++e)
+            {
+              A[d][e] = 0.;
+              for (unsigned int v=0; v<GeometryInfo<dim>::vertices_per_cell; ++v)
+                A[d][e] += vertex[v][d] * TransformR2UInitialGuess<dim>::KA[v][e];
+            }
+        // b = vertex * Kb
+        double b[spacedim];
```

Comment:
why not just `Tensor<1,spacedim>`? Then you can save yourself some of the hand-written loops further below.